### DISCUSSION TITLE
Developing packCells tool to include arbitrary geometry

### DISCUSSION THREAD

**Discussion by mojtabaAmiraslanpour** [UvaCsl/HemoCell#69] [General discussion]
Hello,

I am looking to develop the current packCells tool (which is extraordinary by the way) to include any arbitrary geometry other than a simple box. While I am working on this any helps on where to start is appreciated.

Thank you.

**Comment by mojtabaAmiraslanpour** [comment]
I cannot find the portion of the code dealing with the box boundaries. What I understood from the code by taking a glance at it is that it places all cells in regular positions (which I could not actually find where it sets the position for each cell) and then translates/rotates them later due to penetrations.

**Comment by gzavo** [comment]
Hi @mojtabaAmiraslanpour,

The packCells tool uses periodic boundaries in all 3 dimensions. The cells are positioned randomly, then we compute the overlaps and impose pair forces pushing away the overlapping ellipsoids.

Creating different enclosing boundaries would be very interesting. The main challenge is most likely how to make the collision check with an arbitrary boundary performant.
